jsonschema-core.md

```diff
@@ -1798,24 +1788,26 @@ The minimum number of occurrences is provided by the `minContains` keyword
 within the same schema object as `contains`. If `minContains` is absent, the
 minimum number of occurrences MUST be 1.
 
-Implementations MAY implement the dependency on `minContains` and `maxContains`
-by inspecting their values rather than reading annotations produced by those
-keywords.
-
-This keyword produces an annotation value which is an array of the indexes to
+This keyword produces an annotation value which is an array of the indices for
 which this keyword validates successfully when applying its subschema, in
 ascending order. The value MAY be a boolean `true` if the subschema validates
 successfully when applied to every index of the instance. The annotation MUST be
 present if the instance array to which this keyword's schema applies
 is empty.
 
-This annotation affects the behavior of `unevaluatedItems`.
+The presence of this keyword affects the behavior of
+[`unevaluatedItems`](#unevaluateditems).
+
+Under most circumstances, the `contains` subschema MAY be short-circuited.
+However, for the following cases, the `contains` subschema MUST be applied to
+every array element.
 
-The subschema MUST be applied to every array element even after the first match
-has been found, in order to collect annotations for use by other keywords. This
-is to ensure that all possible annotations are collected.
+- If `unevaluatedItems` appears in any subschema in the dynamic scope that
+  applies to the same instance location, to ensure that all evaluated items are
+  accounted for.
+- When collecting annotations, to ensure that all annotations are found.
 
-## Keywords for Unevaluated Locations
+## Keywords for Unevaluated Locations {#unevaluated}
```

Comment:
..and here?

Reply:
No, this one is defining the anchor.  It needs the `#`.  References use double braces and no `#`, e.g. `{{unevaluated}}`.